src/Storages/Statistics/CountMinSketchStatistics.cpp

```diff
@@ -0,0 +1,100 @@
+#include <Storages/Statistics/CountMinSketchStatistics.h>
+#include <DataTypes/DataTypeLowCardinality.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+
+#if USE_DATASKETCHES
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int ILLEGAL_STATISTICS;
+}
+
+
+CountMinSketchStatistics::CountMinSketchStatistics(const SingleStatisticsDescription & stat_, DataTypePtr data_type_)
+    : IStatistics(stat_)
+    , sketch(HASH_COUNT, BUCKET_COUNT)
+    , data_type(data_type_)
+{
+}
+
+Float64 CountMinSketchStatistics::estimateEqual(const Field & value) const
+{
+    if (auto float_val = IStatistics::getFloat64(value))
+        return sketch.get_estimate(&float_val.value(), 8);
+    if (auto string_val = IStatistics::getString(value))
+        return sketch.get_estimate(string_val->data(), string_val->size());
+    UNREACHABLE();
+}
+
+bool CountMinSketchStatistics::checkType(const Field & f)
+{
+    switch (f.getType())
+    {
+        case Field::Types::Int64:
+        case Field::Types::UInt64:
+        case Field::Types::Float64:
+        case Field::Types::String:
+            return true;
+        default:
+            return false;
+    }
+}
+
+void CountMinSketchStatistics::serialize(WriteBuffer & buf)
+{
+    Sketch::vector_bytes bytes = sketch.serialize();
+    writeIntBinary(static_cast<UInt64>(bytes.size()), buf);
+    buf.write(reinterpret_cast<const char *>(bytes.data()), bytes.size());
+}
+
+void CountMinSketchStatistics::deserialize(ReadBuffer & buf)
+{
+    UInt64 size;
+    readIntBinary(size, buf);
+
+    Sketch::vector_bytes bytes;
+    bytes.reserve(size);
+    buf.readStrict(reinterpret_cast<char *>(bytes.data()), size);
+
+    sketch = datasketches::count_min_sketch<Float64>::deserialize(bytes.data(), size, datasketches::DEFAULT_SEED);
```

Comment:
This used to call `.merge` on the existing sketch. Didn't understand why that is necessary and removed it.